### PR TITLE
add configuration for setting feature gates

### LINKED ISSUE #3804: Allow for feature gates to be configured via the configuration API
**What would you like to be added**:

Feature gates should also be configurable via configuration API.

**Why is this needed**:

I am working on an operator to manage upgrades/configuring of Kueue. Since Kueue deployment uses a configmap I am able to "configure" kueue via changing values in the config map for other personas. 

My hope is to avoid changing the deployment spec of Kueue so I can keep the configuring and upgrades of kueue separate from having to change the deployment.

If users would like to enable feature gates, they would have to modify the CLI arguments of Kueue.

It would be ideal to have a corresponding API for feature gates in the configuration so one can specify either the CLI or the component config.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Allow for feature gates in the configuration API
#### Which issue(s) this PR fixes:

Fixes #3804 

#### Does this PR introduce a user-facing change?

```release-note
User can featureGate in the configuration API.
```

### CODE REVIEW

**Review by tenzen-y — COMMENTED** [review]
API looks good.

File: cmd/kueue/main.go [hunk 1]
```diff
@@ -114,22 +114,37 @@ func main() {
...
+	if err != nil {
+		setupLog.Error(err, "Unable to load the configuration")
+		os.Exit(1)
+	}
+
+	if featureGates != "" && cfg.FeatureGates != nil {
+		err := errors.New("conflicting feature gates detected")
+		setupLog.Error(err, "Unable to decide which features gate to use due to --feature-gates specified at CLI and featureGates specified in the configuration file")
+		// To avoid future bugs, we should fail to start Kueue if feature gates
+		// are specified for both the CLI and the configuration.
```
Comment (mimowo) [inline cmd/kueue/main.go:127]: I think this statement about "avoiding future bugs" is vague, and not visible to the end-users. Also, I think we may want to actually in the future relax this and handle the config map as override. So, I would suggest to drop this comment and update the API field description (as proposed in another comment).

File: apis/config/v1beta1/configuration_types.go [hunk 2]
```diff
@@ -83,6 +83,10 @@ type Configuration struct {
 
 	// Resources provides additional configuration options for handling the resources.
 	Resources *Resources `json:"resources,omitempty"`
+
+	// featureGates is a map of feature names to bools that enable or disable
+	// alpha/experimental feature.
```
Comment (mimowo) [inline apis/config/v1beta1/configuration_types.go:88]:
```suggestion
	// featureGates is a map of feature names to bools that allows to override the
	// default enablement status of a feature. The map cannot be used in conjunction
	// with passing the list of features via the command line argument "--feature-gates"
	// for the Kueue Deployment.
```
I think alpha/experimental is unnecessary and confusing, the map can also be used to disable a beta feature.

**Review by mimowo — COMMENTED** [review]
LGTM overall

File: apis/config/v1beta1/configuration_types.go (hunk 2 above)
Comment (kannon92) [reply apis/config/v1beta1/configuration_types.go]: updated.

File: cmd/kueue/main.go (hunk 1 above)
Comment (kannon92) [reply cmd/kueue/main.go]: updated. PTAL.

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve

**Comment by tenzen-y** [comment]
/release-note-edit
```release-note
Support `.featureGates` field in the configuration API to enable and disable the Kueue features
```

**Pull request merged**